client/src/components/SignIn.tsx

```diff
@@ -95,23 +95,23 @@ function SignIn(props: { handleChange: Function; showAlert: Function }) {
     }
 
     try {
-      const result = await api.signIn(signinForm);
-      if (!result) {
-        throw new Error("Error occurred during sign in.");
-      }
-      if (result.error) {
-        if (!result.error.response) {
+      const { status, data } = await api.signIn(signinForm);
+      if (status < 200 || status >= 300) {
```

Comment:
This type of generic handling is usually bad. I would say status not equal to 204